src/Raven.Client/Documents/Linq/RavenQueryProviderProcessor.cs

```diff
@@ -579,11 +579,22 @@ private string GetPropertyName(string selectPath, ExpressionType type)
                 selectPath += ".Length";
             }
 
-            var propertyName = IndexName == null && _collectionName != null
-                ? QueryGenerator.Conventions.FindPropertyNameForDynamicIndex(typeof(T), IndexName, CurrentPath,
-                    selectPath)
-                : QueryGenerator.Conventions.FindPropertyNameForIndex(typeof(T), IndexName, CurrentPath,
+            string propertyName;
+            if (IndexName == null && _collectionName != null)
+            {
+                propertyName = QueryGenerator.Conventions.FindPropertyNameForDynamicIndex(typeof(T), IndexName, CurrentPath, 
+                    selectPath);
+            }
+            else if (_insideSelect > 0 && QueryGenerator.Conventions.FindProjectedPropertyNameForIndex != null)
+            {
+                propertyName = QueryGenerator.Conventions.FindProjectedPropertyNameForIndex(typeof(T), IndexName, CurrentPath,
```

Comment:
We need to support returning null from 'FindProjectedPropertyNameForIndex' and then fallback to 'FindPropertyNameForIndex' also